### ISSUE TITLE
1.17 Could not pass event InventoryOpenEvent to IllegalStack

### ISSUE THREAD

**Issue by Roshanxy** [dniym/IllegalStack#126]
```
[10:09:24 ERROR]: Could not pass event InventoryOpenEvent to IllegalStack v2.2.2a
java.nio.charset.UnsupportedCharsetException: false
at java.nio.charset.Charset.forName(Charset.java:526) ~[?:?]
at me.dniym.listeners.fListener.onShulkerCheck(fListener.java:1146) ~[?:?]
at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor102.execute(Unknown Source) ~[?:?]
at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.17.jar:git-Purpur-1190]
at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.17.jar:git-Purpur-1190]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.jar:git-Purpur-1190]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.17.jar:git-Purpur-1190]
at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.callInventoryOpenEvent(CraftEventFactory.java:1299) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.server.level.ServerPlayer.openMenu(ServerPlayer.java:1502) ~[?:?]
at net.minecraft.world.level.block.ShulkerBoxBlock.use(ShulkerBoxBlock.java:89) ~[?:?]
at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.use(BlockBehaviour.java:940) ~[?:?]
at net.minecraft.server.level.ServerPlayerGameMode.useItemOn(ServerPlayerGameMode.java:586) ~[?:?]
at net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItemOn(ServerGamePacketListenerImpl.java:1872) ~[?:?]
at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:33) ~[?:?]
at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:9) ~[?:?]
at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:56) ~[?:?]
at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1413) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.server.MinecraftServer.shouldRun(MinecraftServer.java:193) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[?:?]
at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1391) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1384) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[?:?]
at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1360) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1271) ~[patched_1.17.jar:git-Purpur-1190]
at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:322) ~[patched_1.17.jar:git-Purpur-1190]
at java.lang.Thread.run(Thread.java:831) [?:?]
```

**Comment by Roshanxy** [comment]
This happens if I open a shulker in spectator mode

**Comment by dniym** [comment]
Just tried this on a recent spigot build and could not reproduce, are you still getting this error?

**Comment by Roshanxy** [comment]
> Just tried this on a recent spigot build and could not reproduce, are you still getting this error?

Hii yes it doesn't appear any more. So I am guessing it wasn't your code ! But thank you very much for updating it to 1.17 :)

**Comment by Roshanxy** [comment]
Hoi I just got another one but I don't know what caused it. I updated it to 1.17 (The plugin) and use the latest paper version.
```
14:48:09] [Server thread/ERROR]: Could not pass event InventoryOpenEvent to IllegalStack v2.3
java.nio.charset.UnsupportedCharsetException: false
	at java.nio.charset.Charset.forName(Charset.java:526) ~[?:?]
	at me.dniym.listeners.fListener.onShulkerCheck(fListener.java:1151) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor103.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.17.jar:git-Purpur-1251]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.17.jar:git-Purpur-1251]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.jar:git-Purpur-1251]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.17.jar:git-Purpur-1251]
	at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.callInventoryOpenEvent(CraftEventFactory.java:1299) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.server.level.ServerPlayer.openMenu(ServerPlayer.java:1507) ~[?:?]
	at net.minecraft.world.level.block.EnderChestBlock.use(EnderChestBlock.java:88) ~[?:?]
	at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.use(BlockBehaviour.java:940) ~[?:?]
	at net.minecraft.server.level.ServerPlayerGameMode.useItemOn(ServerPlayerGameMode.java:586) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItemOn(ServerGamePacketListenerImpl.java:1872) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:33) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:9) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:56) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1414) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.server.MinecraftServer.shouldRun(MinecraftServer.java:193) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1392) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1385) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1361) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1272) ~[patched_1.17.jar:git-Purpur-1251]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:323) ~[patched_1.17.jar:git-Purpur-1251]
	at java.lang.Thread.run(Thread.java:831) [?:?]

```

I just happen to open up the console and saw this error. I do not know what caused it.

**Comment by dniym** [comment]
it appears you have:  RemoveBooksNotMatchingCharset enabled..   There really shouldn't be any reason for a 1.17 server to have this on as it pertains to the chunk saving dupe.   This should be fixed in every version after 1.15  I'll make the next version of illegalstack turn this off. 

but /istack toggle RemoveBooksNotMatchingCharset should turn this protection off and get rid of this error.
